src/com/magento/idea/magento2plugin/actions/generation/generator/EditEntityActionGenerator.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+package com.magento.idea.magento2plugin.actions.generation.generator;
+
+import com.intellij.openapi.project.Project;
+import com.intellij.psi.PsiDirectory;
+import com.intellij.psi.PsiFile;
+import com.jetbrains.php.lang.psi.elements.PhpClass;
+import com.magento.idea.magento2plugin.actions.generation.data.EditEntityActionData;
+import com.magento.idea.magento2plugin.actions.generation.generator.util.DirectoryGenerator;
+import com.magento.idea.magento2plugin.actions.generation.generator.util.FileFromTemplateGenerator;
+import com.magento.idea.magento2plugin.actions.generation.generator.util.PhpClassGeneratorUtil;
+import com.magento.idea.magento2plugin.indexes.ModuleIndex;
+import com.magento.idea.magento2plugin.magento.files.actions.EditEntityActionFile;
+import com.magento.idea.magento2plugin.magento.packages.HttpMethod;
+import com.magento.idea.magento2plugin.magento.packages.code.BackendModuleType;
+import com.magento.idea.magento2plugin.magento.packages.code.FrameworkLibraryType;
+import com.magento.idea.magento2plugin.util.GetPhpClassByFQN;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+import org.jetbrains.annotations.NotNull;
+
+public class EditEntityActionGenerator extends FileGenerator {
```

Comment:
The generator also looks like doubling. We can use the same generator for all the controllers.

Reply:
@VitaliyBoyko We need different generators because of the different variables used by templates. I believe that we will improve this in the nearest feature.